clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2075,14 +2085,17 @@ private void validateStoreSettingAndPopulate(ControllerClient controllerClient,
     jobSetting.storeStorageQuota = storeResponse.getStore().getStorageQuotaInByte();
 
     // Do not enable for deferred swap or hybrid store
-    if (pushJobSetting.deferVersionSwap || storeResponse.getStore().getHybridStoreConfig() != null) {
+    boolean isDeferredSwap =
+        pushJobSetting.deferVersionSwap && !pushJobSetting.isTargetRegionPushWithDeferredSwapEnabled;
+    if (isDeferredSwap || storeResponse.getStore().getHybridStoreConfig() != null) {
       LOGGER.warn(
           "target region is not available for {} as it hybrid or deferred version swap enabled.",
           jobSetting.storeName);
       jobSetting.isTargetedRegionPushEnabled = false;
     }
 
-    if (jobSetting.isTargetedRegionPushEnabled && jobSetting.targetedRegions == null) {
+    if ((jobSetting.isTargetedRegionPushEnabled || jobSetting.isTargetRegionPushWithDeferredSwapEnabled)
```

Comment:
Also can we validate we do not allow both `isTargetedRegionPushEnabled` and `isTargetRegionPushWithDeferredSwapEnabled` to be enabled ?

Reply:
Also add some unit tests to not allow combinations of these configs